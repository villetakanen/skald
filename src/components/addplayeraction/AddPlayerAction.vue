<template>
  <v-dialog
    v-model="dialog"
    width="500"
    >
    <template v-slot:activator="{ on, attrs }">
      <v-btn
        color="primary"
        v-bind="attrs"
        v-on="on"
        >{{$t('view_gm_players.addPlayerToSite')}}</v-btn>
    </template>

    <v-card>
      <v-card-title>
        {{$t('view_gm_players.addPlayerToSite')}}
      </v-card-title>

      <v-card-text>
        <v-select
          v-model="selectedUser"
          :items="allUserNicks"></v-select>
      </v-card-text>

      <v-divider></v-divider>

      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn
          @click="addPlayer(selectedUser)"
          :disabled="!selectedUser"
          color="primary"
          >{{$t('view_gm_players.addPlayerToSiteButton')}}</v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script lang="ts">
import Vue from 'vue'
import VueCompositionApi, { defineComponent, onMounted, ref } from '@vue/composition-api'
import firebase from 'firebase/app'
import 'firebase/firestore'

Vue.use(VueCompositionApi)

interface profile {
  uid: string
  nick: string
}

export default defineComponent({
  props: {
    siteid: {
      type: [String],
      required: true
    }
  },
  setup (props) {
    const dialog = ref(false)
    const selectedUser = ref('')
    const userArray: profile[] = []
    const userNickArray: string[] = []
    onMounted(() => {
      const db = firebase.firestore()
      const playerRef = db.collection('profiles')
      playerRef.get().then((querySnapshot) => {
        querySnapshot.forEach((doc) => {
          userArray.push({ uid: doc.id, nick: doc.data().nick })
          userNickArray.push(doc.data().nick)
        })
      })
    })
    const allUserNicks = ref(userNickArray)
    const allUsers = ref(userArray)

    const addPlayer = (userNick: string) => {
      const db = firebase.firestore()
      const playersRef = db.collection('sites').doc(props.siteid).collection('players')
      const player = allUsers.value.find((user) => { return user.nick === userNick })
      playersRef.doc(player?.uid).set({ nick: userNick, tags: [] }).then(() => {
        dialog.value = false
      })
    }

    return { allUserNicks, allUsers, addPlayer, dialog, selectedUser }
  }
})
</script>

<style lang="scss" scoped>
.AddPlayersAction{
  display: grid;
  grid-template-columns: auto auto;
  .AddPlayersAction-column1{
    padding-right: 4px;
    justify-self: stretch;
  }
  .AddPlayersAction-column2{
    padding-left: 4px;
    justify-self: start;
    align-self: center;
  }
}
</style>
