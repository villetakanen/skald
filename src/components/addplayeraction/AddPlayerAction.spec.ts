import { shallowMount } from '@vue/test-utils'
import AddPlayerAction from './AddPlayerAction.vue'

describe('Mounted App', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(AddPlayerAction, {
      mocks: {
        $t: () => {}
      },
      propsData: {
        siteid: 'skald'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
