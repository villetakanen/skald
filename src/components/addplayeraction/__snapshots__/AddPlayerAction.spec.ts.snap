// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted App renders correctly 1`] = `
<div
  class="AddPlayersAction"
>
  <div
    class="AddPlayersAction-column1"
  >
    <v-select-stub
      appendicon="$dropdown"
      backgroundcolor=""
      clearicon="$clear"
      errorcount="1"
      errormessages=""
      itemcolor="primary"
      itemdisabled="disabled"
      items=""
      itemtext="text"
      itemvalue="value"
      loaderheight="2"
      menuprops="[object Object]"
      messages=""
      nodatatext="$vuetify.noDataText"
      rules=""
      successmessages=""
      type="text"
      value=""
      valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date && b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}"
    />
  </div>
   
  <div
    class="AddPlayersAction-column2"
  >
    <v-btn-stub
      activeclass=""
      tag="button"
      type="button"
    >
      Add a Player
    </v-btn-stub>
  </div>
</div>
`;
